test/as_buffer_async_pubsub_1.cpp

```diff
@@ -47,6 +47,7 @@ BOOST_AUTO_TEST_CASE( pub_qos0_sub_qos0 ) {
                     pid_sub = c->async_subscribe(
                         as::buffer(*topic1),
                         MQTT_NS::qos::at_most_once,
+                        topic1,
                         [topic1](boost::system::error_code const&) {}
```

Comment:
This is an example of the capture code.

Reply:
Updated to remove lambda-capture variables.